lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1465,6 +1469,19 @@ def configure_disks(self, vm_obj):
                 diskspec = self.device_helper.create_scsi_disk(scsi_ctl, disk_index)
                 disk_modified = True
 
+            if 'disk_mode' in expected_disk_spec:
+                disk_mode = expected_disk_spec.get('disk_mode', 'persistent').lower()
+                valid_disk_mode = ['persistent', 'independent_persistent', 'independent_nonpersistent']
```

Comment:
We should plan a cleanup of the module signature and start using sub options, that way, something like this can be captured early